### PR TITLE
psfex 3.17.1 (new formula)

### PR DESCRIPTION
**Pull request by wschoenell**
Adds PSFex software recipe. PSFEx  (“PSF Extractor”) extracts models of the Point Spread Function (PSF) from FITS images processed with SExtractor and measures the quality of images.

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/psfex.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class Psfex < Formula
+  homepage "https://www.astromatic.net/software/psfex"
+  url "https://www.astromatic.net/download/psfex/psfex-3.17.1.tar.gz"
+  sha256 "53f1b449ab7da7e6e0a989c41b82885f52c8f08270ceb4378bb1ec7ef754af89"
+
+  depends_on "fftw"
+  depends_on "plplot"
+  depends_on "autoconf" => :build
```
Comment (javian) [inline Formula/psfex.rb]: `:build` options should be listed first in the list of dependencies

File: Formula/psfex.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  # this patches make the changes needed to compile with the Accelerate
+  # framework for linear algebra routines. Based on sextractor.rb recipe.
+  patch do
```
Comment (javian) [inline Formula/psfex.rb]: Has this patch been submitted upstream ?

Comment (fxcoudert) [reply Formula/psfex.rb]: If the software builds without patch using `openblas`, we might opt for that. Accelerate has issues.

File: Formula/psfex.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  def install
+    system "autoconf"
+    system "autoheader"
+    system "./configure", "--prefix=#{prefix}", "--enable-accelerate"
+    system "make"
+    system "make", "check"
+    system "make", "install"
+  end
+end
```
Comment (fxcoudert) [inline Formula/psfex.rb]: We need a test that exercises the some of the functionality of the app.

**Comment by wschoenell** [comment]
Thanks for your replies. I just opened an issue upstream astromatic/psfex#7 to have a release with OpenBLAS support, so it eliminates the necessity of a patch. I will check the other issues in the next commit.

**Comment by fxcoudert** [comment]
@wschoenell thanks, I am going to close this for now. Please ask here for it to be reopened, or open a new PR when upstream has a new release.

**Pull request closed without merging**